pkg/comp-functions/functions/common/maintenance.go

```diff
@@ -0,0 +1,206 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	"github.com/vshn/appcat/pkg/comp-functions/runtime"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"regexp"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Maintenance contains data for maintenance k8s resource generation
+type Maintenance struct {
+	InstanceNamespace string
+	MainRole          string
+	Service           string
+	Resource          client.Object
+	Ctx               context.Context
+	Iof               *runtime.Runtime
+	Schedule          vshnv1.VSHNDBaaSMaintenanceScheduleSpec
+	PolicyRules       []rbacv1.PolicyRule
+	ExtraEnvs         []corev1.EnvVar
+	ExtraResFuncs     []ExtraResourceFunctions
```

Comment:
Nitpick: Not sure if I'm a fan of how the `ExtraResFuncs` work currently, because they can easily be abused to do pretty much anything..

Reply:
Do you have an idea how I could create extra resources which I need for postgres? I could also leave it in the postgres function but I would prefer to pass it in here somehow.

Reply:
Maybe instead of providing arbitrary functions it could only take a `[]client.Object`?

Reply:
Check it now